explainer.md

```diff
@@ -60,33 +60,27 @@ On platforms that only use file extensions to describe file types, user agents c
 
 The user can right click on CSV or SVG files in the operating system's file browser, and choose to open the files with the Grafr web application. (This option would only be presented if Grafr has been [installed](https://w3c.github.io/manifest/#installable-web-applications).).
 
-Choosing to open the file would create a new top level browsing context, navigating to '{origin}{action}'. Assuming the user opened `graph.csv` in Grafr the URL would be `https://grafr.com/open-csv/`. When the `load` event is fired, an `fileHandles` will be available in the `launchParams` property on the global object.
+Choosing to open the file would create a new top level browsing context, navigating to '{origin}{action}'. Assuming the user opened `graph.csv` in Grafr the URL would be `https://grafr.com/open-csv/`.
 
-> Note: The `launchParams` property is discussed in more detail in the [Launch Events](https://github.com/WICG/sw-launch) explainer.
+To access launched files, a site should subscribe to a `launches` object attached to `window`. The `next` function is a called once for each launch. Launches are cached (in order) until they are consumed by a subscriber. This ensures that we avoid any races when attaching an event listener.
```

Comment:
1. I would prefer to call this `launchQueue` rather than `launches`.
2. s/The `next` function/The `next` method
3. s/cached/queued (this is not a cache)
4. s/a subscriber/the subscriber (we shouldn't allow multiple subscribers)
5. I think we should avoid too closely matching the [observable](https://github.com/tc39/proposal-observable) proposal, in fact should avoid clashing with their names. An attempt to match their semantics (possibly with the hope of one day being able to migrate this to literally being an Observable) is probably a mistake, since we might violate their semantics in some subtle way or they may change their proposal to be different to ours. So we should probably avoid the names `subscribe` and `next`. `next` can be named `onLaunch`. Not sure what to call `subscribe`. Maybe just simplify the API so it doesn't take a class:

```js
launchQueue.setLaunchObserver(launchParams => {
   ...
});
```

That makes it clear that a) this is not an event listener, b) this is not an Observable, c) you are setting a single callback, not adding to a possible list of many callbacks.

Reply:
Changed to
```js
launchQueue.setConsumer(launchParams => {
    ...
});
```